docs/library/micropython.rst

```diff
@@ -147,3 +147,66 @@ Functions
 
    There is a finite queue to hold the scheduled functions and `schedule()`
    will raise a `RuntimeError` if the queue is full.
+
+Classes
+-------
+
+.. class:: RingIO(size)
+.. class:: RingIO(buffer)
+   :noindex:
+
+   Provides a fixed-size ringbuffer for bytes with a stream interface. Can be
+   considered like a fifo queue variant of `io.BytesIO`.
+
+   When created with integer size a suitable buffer will be allocated.
+   Alternatively a `bytearray` or similar buffer protocol object can be provided
+   to the constructor for in-place use.
+
+   The classic ringbuffer algorithm is used which allows for any size buffer
+   to be used however one byte will be consumed for tracking. If initialised
+   with an integer size this will be accounted for, for example ``RingIO(16)``
+   will allocate a 17 byte buffer internally so it can hold 16 bytes of data.
+   When passing in a pre-allocated buffer however one byte less than its
+   original length will be available for storage, eg. ``RingIO(bytearray(16))``
+   will only hold 15 bytes of data.
```

Comment:
Maybe worth mentioning that reading and writing are thread and IRQ safe when used separately?